packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -0,0 +1,1277 @@
+        _dragging = dragging;
+      });
+    }
+  }
+  bool _dragging = false;
+
+  @override
+  void initState() {
+    _listState = SliverReorderableList.of(context);
+    _listState._registerItem(this);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _offsetAnimation?.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(covariant _ReorderableItem oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.key != widget.key || oldWidget.index != widget.index) {
+      _listState._unregisterItem(oldWidget.index, this);
+      _listState._registerItem(this);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_dragging) {
+      return const SizedBox();
+    }
+    return Transform(
+      transform: Matrix4.translationValues(offset.dx, offset.dy, 0.0),
+      child: widget.child,
+    );
+  }
+
+  @override
+  void deactivate() {
+    _listState._unregisterItem(index, this);
+    super.deactivate();
+  }
+
+  Offset get offset {
+    if (_offsetAnimation != null) {
+      final double animValue = Curves.easeInOut.transform(_offsetAnimation!.value);
+      return Offset.lerp(_startOffset, _targetOffset, animValue)!;
+    }
+    return _targetOffset;
+  }
+
+  void updateForGap(int gapIndex, double gapExtent, bool animate, bool reverse) {
+    final Offset newTargetOffset = (gapIndex <= index)
+        ? _extentOffset(reverse ? -gapExtent : gapExtent, _listState._scrollDirection)
+        : Offset.zero;
+    if (newTargetOffset != _targetOffset) {
+      _targetOffset = newTargetOffset;
+      if (animate) {
+        if (_offsetAnimation == null) {
+          _offsetAnimation = AnimationController(
+            vsync: _listState,
+            duration: const Duration(milliseconds: 250),
+          )
+            ..addListener(_update)
+            ..addStatusListener((AnimationStatus status) {
+              if (status == AnimationStatus.completed) {
+                _startOffset = _targetOffset;
+                _offsetAnimation!.dispose();
+                _offsetAnimation = null;
+              }
+            })
+            ..forward();
+        } else {
+          _startOffset = offset;
+          _offsetAnimation!.forward(from: 0.0);
+        }
+      } else {
+        if (_offsetAnimation != null) {
+          _offsetAnimation!.dispose();
+          _offsetAnimation = null;
+        }
+        _startOffset = _targetOffset;
+      }
+      _update();
+    }
+  }
+
+  void resetGap() {
+    if (_offsetAnimation != null) {
+      _offsetAnimation!.dispose();
+      _offsetAnimation = null;
+    }
+    _startOffset = Offset.zero;
+    _targetOffset = Offset.zero;
+    _update();
+  }
+
+  Rect targetGeometry() {
+    final RenderBox itemRenderBox = context.findRenderObject()! as RenderBox;
+    final Offset itemPosition = itemRenderBox.localToGlobal(Offset.zero) + _targetOffset;
+    return itemPosition & itemRenderBox.size;
+  }
+
+  void _update() {
```

Comment:
This appears to be the only method with a _foo private name. Maybe it should get a regular name, like `rebuild()`